File: 01. Lenguajes de proposito general/02. Programación básica/07. Funciones/06. Con paramatros y con retorno multiple/C/Con paramatros y con retorno multiple.c

// Estructura de retorno
struct Return {
    tipo return1;
    tipo return2;
    tipo returnN;
};


// Creación con parametros limitados
Return funcion(tipo var1, tipo var2, tipo varN) {
    // Contenido
    return (Return) {valor1, valor2, valorN};
}

// Creación con parametros ilimitados
Return funcion(tipo variables, ...) {
    // Contenido
    return (Return) {valor1, valor2, valorN};
}

// Creación con parametros por defecto (Los valores por defecto de pueden omitir en el llamado)
NO SOPORTADO

int main() {
    
    // Llamado
    /*Contexto*/ funcion(valor1, valor2, valorN) /*Contexto*/

    // Llamado con asignación
    Return variables = funcion(valor1, valor2, valorN);
    tipo varible1 = variables.return1;
    tipo varible2 = variables.return2;
    tipo varibleN = variables.returnN;

    // Eliminación
    NO SOPORTADO (SE ELIMINAN AL SALIR DE LOS CORCHETES CONTENEDORES)

}

File: 01. Lenguajes de proposito general/01. Programación básica/08. Funciones/06. Con paramatros y con retorno multiple/Java/Con paramatros y con retorno multiple.java

package prueba;

public class Prueba {

    // Creación con parametros limitados
    public static Object[] funcion(tipo var1, tipo var2, tipo varN) {
        // Contenido
        return new Object[]{valor1, valor2, valorN};
    }

    // Creación con parametros ilimitados
    public static Object[] funcion(tipo ...variables) {
        // Contenido
        return new Object[]{valor1, valor2, valorN};
    }

    // Creación con parametros por defecto (Los valores por defecto de pueden omitir en el llamado)
    NO SOPORTADO

    public static void main(String[] args) {
        
        // Llamado
        /*Contexto*/ funcion(valor1, valor2, valorN) /*Contexto*/

        // Llamado con asignación
        Object[] variables = funcion(valor1, valor2, valorN);
        tipo varible1 = (tipo) variables[0];
        tipo varible2 = (tipo) variables[1];
        tipo varibleN = (tipo) variables[2];

        // Eliminación
        NO SOPORTADO (EL GARBAGE COLLECTOR LA ELIMINA AUTOMATICAMENTE CUANDO YA ES INACCESIBLE)

    }
    
}